.github/workflows/pylint.yml

```diff
@@ -21,11 +21,19 @@ jobs:
         python get-pip.py
         pip install pylint==1.1.0
         pip install astroid==1.2
-        
+
     - name: Run Pylint Checks
       run: |
-        pylint --rcfile=.pylintrc convert2rhel/
-        
+         function interpreted_pylint() {
+            local es=0
+            pylint --rcfile=.pylintrc convert2rhel/; es=$?
+            case $es in
+                 1|2|3)  return $es ;;
```

Comment:
I'm not familiar with bitwise arithmetic so can't check that part but otherwise it looks like it should work.

Just a nitpick, better coding practice is to use `local` when introducing new vars, and ideally just declare them (perhaps with default values and/or descriptive in-line comments) in function header.  Doesn't matter too much in a one-off throw-away function, though. :)

Anyway, if pylint has this compex way of reporting result, it seems worth considering writing the wrapper as separate binary that does all the arithmetic and perhaps shows real summary of what pylint wanted to say.  (Or see if such tool already exists.)  I could help write one in Bash, OTOH you would not need to use Bash at all at that point.

Reply:
Thanks for the remarks. It still doesn't work: ` ##[error]Process completed with exit code 16.` In the output I see `shell: sh -e {0}`. I don't know what that means. The man page for sh is so huge and `-e` is mentioned there hundreds of time. I think we'll just drop this effort and try to fix all the messages.

Reply:
Or, @fellipeh, please create a task for the infra team so they make sure the checks fail only if there are fatal and error messages. Unfortunately in the mean time the check will not be useful for us at all because it will be failing all the time.

Reply:
Not sure about `{0}` (probably irrelevant to shell) but `-e` would explain a lot.  It enables the mode when shell [will exit on first (untreated) non-zero](https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html) exit status.  It's actually common for pipeline shells to run in this mode.

You can disable it by adding `set +e`.

(Something like this should also work

    pylint --yada-yada .. || es=$?

but you would need to make sure `es` is initialzed to 0.)